system-int-test/test/cmr/system_int_test/ingest/generics_test.clj

```diff
@@ -0,0 +1,204 @@
+            expected2 "X"
+            actual2 (gdocs/get-sub-concept-type-concept-id-prefix :fake "A.B.C")]
+        (is (= expected1 actual1) "was not able to find GRD")
+        (is (= expected2 actual2) "was not able to default to X"))
+
+      (dev-sys-util/eval-in-dev-sys `(config/set-approved-pipeline-documents! `original-setting)))))
+
+(deftest test-generic-CRUD
+  "Test that a Generic can be walked through all the CRUD actions using the ingest
+   interface. Use the same native-id for all these steps"
+
+  (let [original-setting (config/approved-pipeline-documents)
+        native-id (format "Generic-Test-%s" (UUID/randomUUID))
+        generic-requester (partial generic-request "PROV1" native-id)
+        good-generic-requester (partial generic-requester grid-good)]
+
+    (testing "send a good document with config set that does not include grid"
+      (dev-sys-util/eval-in-dev-sys `(config/set-approved-pipeline-documents! {:fake ["u.r.a.b"]}))
+      (let [result (good-generic-requester :post)]
+        (is (= 422 (:status result)) "The HTTP status code is not correct")
+        (is (= "application/json" (get-in result [:headers "Content-Type"]))
+            "The content type is not correct")
+        (is (= "{\"errors\":[\"The [:grid] schema on version [0.0.1] is not an approved schema. This record cannot be ingested.\"]}"
+               (:body result))
+            "The body is not correct")))
+
+    (testing "CREATE a document with a valid config set that includes grid"
+      (dev-sys-util/eval-in-dev-sys `(config/set-approved-pipeline-documents! {:grid ["0.0.1"]}))
+      (let [expected {:Info {:native-id native-id
+                             :document-name "Grid-A7-v1"
+                             :provider-id "PROV1"
+                             :schema "grid"
+                             :format "grid"
+                             :mime-type "application/grid;version=0.0.1"
+                             :revision-id 1
+                             :user-id "ECHO_SYS"}
+                      :Metadata grid-good}
+
+            create-result (good-generic-requester :post)
+            create-response (:body create-result)
+
+            read-result (generic-requester)
+            read-response (:body read-result)
+
+            actual (json/parse-string read-response true)
+            concept-id (get-in actual [:Info :concept-id])
+            normalised (-> actual
+                           ;; these fields are complicated to test, do so another way
+                           (cutil/dissoc-in [:Info :concept-id])
+                           (cutil/dissoc-in [:Info :revision-date])
+                           (cutil/dissoc-in [:Info :created-at]))]
+        ;; test that the create was a success
+        (is (= 200 (:status create-result)) "The HTTP status code from create is not correct")
+        (is (= "" create-response) "Create returned content when it should not have")
```

Comment:
You are testing getting nothing back.  When we create a document we need to get back a response to the user.  Just like we do for the other concepts like collection.  The end user experience should be consistent when working with different concepts.  Some concept ingest response shouldn't be different from others.

Reply:
I was not sure what you wanted to be returned, but I have changed this to match the concepts.